packages/components/src/components/dropdown-list/dropdown-list.tsx

```diff
@@ -1,9 +1,47 @@
-import classNames from 'classnames';
 import React from 'react';
+import classNames from 'classnames';
 import ReactDOM from 'react-dom';
 import { CSSTransition } from 'react-transition-group';
-import PropTypes from 'prop-types';
 import ThemedScrollbars from '../themed-scrollbars/themed-scrollbars';
+import { ResidenceList } from '@deriv/api-types';
+
+type TItem = string & {
+    component?: React.ReactNode;
+    group?: string;
+} & ResidenceList[0];
```

Comment:
@niloofar-deriv This type seems too loose and complicated, Should it be `&` or should it be `|`? 🤔

Reply:
This should be &, I'll change the structure a little bit to reduce confusion